src/util/adb.js

```diff
@@ -213,6 +213,54 @@ export default class ADBUtils {
 
     return artifactsDir;
   }
+  fun(x: string) {
+    return x;
+  }
+  async checkArtifacts(deviceId: string): Promise<boolean> {
+    const {adbClient} = this;
+    await wrapADBCall(async () => {
+      await adbClient.readdir(deviceId, '/sdcard')
+        .then(function(dirs) {
+          // Synchronous, so we don't have to care about returning at the
+          // right time
+          dirs.forEach(function(dir) {
+            if (dir.isDir()) {
+              if (dir.name.startsWith('web-ext-artifacts-')) {
+                return false;
+              }
+            }
+          });
+        });
+    });
+    return true;
+  }
+  async clearArtifacts(deviceId: string): Promise<boolean> {
+    const {adbClient} = this;
+    const directory = [];
+    await wrapADBCall(async () => {
+      await adbClient.readdir(deviceId, '/sdcard')
+        .then(function(dirs) {
+          // Synchronous, so we don't have to care about returning at the
+          // right time
+          dirs.forEach(function(dir) {
+            if (dir.isDir()) {
+              if (dir.name.startsWith('web-ext-artifacts-')) {
+                log.debug(
+                  `Removing ${dir.name} artifacts directory on ${deviceId}`
+                );
+                directory.push(dir.name);
+              }
+            }
+          });
+        });
+    });
+    for (const names of directory) {
+      await this.runShellCommand(deviceId, [
+        'rm', '-rf', names,
+      ]);
+    }
+    return true;
+  }
```

Comment:
this `checkArtifacts` and `clearArtifacts` methods are pertty similar to each other, it looks like they could be merged into a single method (which takes an additional boolean parameter that would make it **"check and clear"** the artifacts directories vs. just **checking** them)